heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java

```diff
@@ -280,6 +282,11 @@ private SchedulerUtils() {
     commands.add(createCommandArg(ExecutorFlag.HealthManagerMode, healthMgrMode));
     commands.add(createCommandArg(ExecutorFlag.HealthManagerClasspath,
         Context.healthMgrClassPath(config)));
+    if (freePorts.size() > 9) {
```

Comment:
Why 9? Suggest to make a const for it with a name.

Also I am wondering if it could be useful to have an else {} block with a log?

Reply:
This is because ports are pass in as positional arguments.  There are 9 ports prior to the debugging ports which could be more than one port.  

This PR should address the positional arguments issue:

https://github.com/twitter/heron/pull/2516

Reply:
I see. Feels a little scary for future maintenance, especially for adding or removing ports. I looked at the other PR briefly but don't have good idea about the typical use cases yet. A few questions I am wondering so far:

- is this map necessary? can we limit the jvm debugger to be one at a time? or it is normal to have multiple debuggers in jvm world?
- could it be better to refactor all ports to be a port map instead of creating a map for debugging port and leave others as is?

Reply:
@nwangtw 
Your first question:

The reason why we pass multiple debugger ports is because the heron_executor.py can potentially starts up multiple heron instances based on the physical plan thus we need multiple debugger port assignments to be passed to it.

Second question:

That is essentially what #2516 does.  Puts all ports in a map